addons/sale_pdf_quote_builder/models/product_document.py

```diff
@@ -24,9 +24,26 @@ class ProductDocument(models.Model):
              "header pages and the quote table. ",
     )
 
-    @api.constrains('attached_on', 'datas')
+    @api.onchange('attached_on', 'type')
+    def _onchange_attached_on(self):
+        for record in self:
+            if record.attached_on == 'inside' and record.type != 'binary':
+                return {
+                    'warning': {
+                        'title': _("Invalid Document Type"),
+                        'message': _(
+                            "When attaching inside a quote, the document must be a file, not a URL."
+                        ),
+                    }
+                }
```

Comment:
Spoke about it with dala: as there is the constraint anyway, no need to add the warning, especially since we didn't add a warning for non pdf datas, just the warning. Adding both would be duplicated code for nothing. =]